docs/hub/searching-the-hub.md

```diff
@@ -0,0 +1,189 @@
+- `filter`
+- `author`
+- `search`
+- ...
+
+Two of these parameters make some sense (`author` and `search`), but what about `filter`? Let's dive into a few helpers quickly and revisit that question.
+
+## Search Parameters
+
+The `huggingface_hub` provides a user-friendly interface to know what exactly can be passed into this `filter` parameter through the `ModelSearchArguments` and `DatasetSearchArguments` classes:
+
+
+```python
+>>> from huggingface_hub import ModelSearchArguments, DatasetSearchArguments
+
+>>> model_args = ModelSearchArguments()
+>>> dataset_args = DatasetSearchArguments()
+```
+
+These are nested namespace objects that have **every single option** available on the Hub, with tab-completion, that will return what should be passed to `filter`. 
+
+Let's pose a problem that would be complicated to solve without access to this information:
+> I want to search the Hub for all PyTorch models that can do Text Classification, and trained on the `glue` dataset
+
+If we check what is available in our `model_args` by checking it's output in a Jupyter cell, we will find:
+
+
+```python
+>>> model_args
+```
+
+
+
+
+    Available Attributes or Keys:
+     * author
+     * dataset
+     * language
+     * library
+     * license
+     * model_name
+     * pipeline_tag
+
+
+
+It has a variety of attributes or keys available to us. This is because it is both an object and a dictionary, so we can either do `model_args["author"]` or `model_args.author`. For this tutorial we will follow the latter format.
+
+Our first criteria is we want all PyTorch models. This would be found under the `library` attribute, so let's see if it is there:
+
+
+```python
+>>> model_args.library
+```
+
+
+
+
+    Available Attributes or Keys:
+     * AdapterTransformers
+     * Asteroid
+     * ESPnet
+     * Flair
+     * JAX
+     * Joblib
+     * Keras
+     * ONNX
+     * PyTorch
+     * Pyannote
+     * Rust
+     * Scikit_learn
+     * SentenceTransformers
+     * Stanza
+     * TFLite
+     * TensorBoard
+     * TensorFlow
+     * TensorFlowTTS
+     * Timm
+     * Transformers
+     * allennlp
+     * fastText
+     * fastai
+     * spaCy
+     * speechbrain
+
+
+
+We can see that it is under `PyTorch`. So what we want to use is `model_args.library.PyTorch`:
+
+
+```python
+>>> model_args.library.PyTorch
+```
+
+
+
+
+    'pytorch'
+
+
+
+Below is a graphic repeating the process for finding both our `Text Classification` requirement, as well as our `glue` requirement:
+
+![](../assets/hub/search_text_classification.gif)
+
+![](../assets/hub/search_glue.gif)
+
+Now that we have the pieces, we just need to combine them all together for something the api can use, through the `ModelFilter` and `DatasetFilter` classes. These take those outputs we just had, and transform them into something the API can use conviently:
+
+
+```python
+>>> from huggingface_hub import ModelFilter, DatasetFilter
```

Comment:
```suggestion
>>> from huggingface_hub import ModelFilter
```

Removed as it was not used

Reply:
Opting to keep it in for a Dataset example at the end